### PR TITLE
Add `pynvml` as a dependency for `dask-cudf`

### PR DESCRIPTION
**Pull request by pentschev**
## Description
https://github.com/rapidsai/cudf/pull/17250 started using `pynvml` but did not add the proper dependency, this change fixes the missing dependency.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by rjzamora** [comment]
>Do we want to drop this try...except... then?

The `ImportError` shouldn't happen anymore, but the `ValueError` could happen if `CUDA_VISIBLE_DEVICES` isn't set correctly (e.g. if Dask-CUDA isn't in use).

**Comment by pentschev** [comment]
Removed the `ImportError` catch in https://github.com/rapidsai/cudf/pull/17386/commits/da924080286ab236025e46392f47969645b33d64

**Review by galipremsagar — APPROVED** [review]

File: conda/recipes/dask-cudf/meta.yaml [hunk 1]
```diff
@@ -43,6 +43,7 @@ requirements:
   run:
     - python
     - cudf ={{ version }}
+    - pynvml >=11.4.1,<12.0.0a0
```
Comment (jakirkham) [inline conda/recipes/dask-cudf/meta.yaml:46]: Pushed this small change to include `pynvml` in the `dask-cudf` Conda package. Hope that is ok 🙂

**Review by jakirkham — APPROVED** [review]
Thanks Peter and Rick! 🙏

**Pull request merged**